jobrunner/executors/local.py

```diff
@@ -72,17 +72,34 @@ def get_log_dir(job_id):
     return config.JOB_LOG_DIR / month_dir / container_name(job_id)
 
 
-def read_job_metadata(job_id):
+def read_job_metadata(job_id, redact=True):
```

Comment:
Rather than having a flag here I think it might be both simpler and clearer to factor this as:
```py
def read_job_metadata(job_id):
    metadata = read_job_metadata_unredacted(job_id)
    return redact_job_metadata(metadata)
```
Then code which needs access to the unredacted metadata can use the other function.

Reply:
Hmm, we need the unredacted data sometimes though? Like, to know which files to copy?

I think we should only redact at the system boundary - when we send it to the controller?

I think its fine for JobStatus.results to be the unredacted version - the agent loop may need to use the unredacted info, even it doesnt right now?

But agent.main.update_job_task can do the redaction before passing it up to the controller? That way, we *ensure* that the data is redacted. Currently, it's possible make mistakes in future an accidentally send an unredacted results up to the controller?